logger.go

```diff
@@ -104,6 +104,17 @@ func Scoped(scope string, description string) Logger {
 	return adapted.Scoped(scope, description).With(otelfields.AttributesNamespace)
 }
 
+// NoOp returns a no-op Logger that can never produce any output. It can be safely created
+// before initialization. Use sparingly, and do not use with the intent of replacing it
+// post-initialization.
```

Comment:
Let's explicitly specify the intended use case here

Reply:
I'm a bit hesitant to mention `TestContext` specifically here - this is still a standalone package, and the `TestContext` mentioned is part of `internal/observation` in `sourcegraph/sourcegraph`. I hope the combination of a code insight/monitor + the docstring indicating that this should not be used as swappable logging is sufficient.

Additionally, there might be an arguable use case for a swappable logging implementation - for example, if a struct has opt-in logging, it can start with a NoOp logger and allow users to provide their own loggers as desired, so maybe we don't want to be too strict here

Reply:
You're being both good cop and bad cop here 😆 DON'T USE THIS LOGGER<sup>unless you want to🥰</sup>